packages/atomic/src/components/search/atomic-field-condition/atomic-field-condition.ts

```diff
@@ -0,0 +1,103 @@
+import type {Result, ResultTemplateCondition} from '@coveo/headless';
+import {ResultTemplatesHelpers} from '@coveo/headless';
+import {type CSSResultGroup, css, html, LitElement} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
+import {
+  makeDefinedConditions,
+  makeMatchConditions,
+} from '@/src/components/common/template-controller/template-utils';
+import type {Bindings} from '@/src/components/search/atomic-search-interface/interfaces';
+import {createResultContextController} from '@/src/components/search/result-template-component-utils/context/result-context-controller';
+import {bindings} from '@/src/decorators/bindings';
+import {errorGuard} from '@/src/decorators/error-guard';
+import type {InitializableComponent} from '@/src/decorators/types';
+import {LightDomMixin} from '@/src/mixins/light-dom';
+import {mapProperty} from '@/src/utils/props-utils';
+
+/**
+ * The `atomic-field-condition` component renders its children only when all of the conditions specified through its props are satisfied.
+ * The condition properties can be based on any top-level result property of the `result` object, not restricted to fields (e.g., `isRecommendation`).
+ *
+ * @slot default - The content to display if the conditions are met.
+ */
+@customElement('atomic-field-condition')
+@bindings()
+export class AtomicFieldCondition
+  extends LightDomMixin(LitElement)
+  implements InitializableComponent<Bindings>
+{
+  @state() bindings!: Bindings;
+  @state() error!: Error;
+
+  private resultController = createResultContextController(this);
+
+  /**
+   * A condition that is satisfied when the specified field is defined on a result (e.g., `if-defined="author"` is satisfied when a result has the `author` field).
+   */
+  @property({type: String, attribute: 'if-defined'}) ifDefined?: string;
+
+  /**
+   * A condition that is satisfied when the specified field is not defined on a result (e.g., `if-not-defined="author"` is satisfied when a result does not have the `author` field).
+   */
+  @property({type: String, attribute: 'if-not-defined'}) ifNotDefined?: string;
+
+  /**
+   * A condition that is satisfied when the specified field matches one of the specified values on a result (e.g., `must-match-filetype="pdf,docx"` is satisfied when a result has a filetype of either pdf or docx).
+   * @type {Record<string, string[]>}
+   */
+  @mapProperty({splitValues: true, attributePrefix: 'must-match'})
+  mustMatch!: Record<string, string[]>;
+
+  /**
+   * A condition that is satisfied when the specified field does not match any of the specified values on a result (e.g., `must-not-match-filetype="pdf"` is satisfied when a result does not have a filetype of pdf).
+   * @type {Record<string, string[]>}
+   */
+  @mapProperty({splitValues: true, attributePrefix: 'must-not-match'})
+  mustNotMatch!: Record<string, string[]>;
+
+  static styles: CSSResultGroup = css`
+    :host {
+      max-width: 100%;
```

Comment:
Mm why is this necessary? We don't have this on atomic-product-field-condition.ts